### PR TITLE
UnityHttpRequest Option

### PR DESCRIPTION
**Pull request by nitreo**
- added UnityHttpRequest option
- test runner "Clear Logs" quick stub to avoid expeptions

### CODE REVIEW

File: targets/unity-v2/Testing/Tests/Shared/Uunit/Editor/UUnitTestRunner.cs [hunk 1]
```diff
@@ -1,5 +1,8 @@
 using System;
+using System.CodeDom;
 using System.Reflection;
+using NUnit.Framework.Constraints;
+using PlayFab.ClientModels;
```
Comment (pgilmorepf) [inline targets/unity-v2/Testing/Tests/Shared/Uunit/Editor/UUnitTestRunner.cs]: Don't add a bunch of unused usings.

File: targets/unity-v2/templates/PlayFabSettings.cs.ejs [hunk 2]
```diff
@@ -7,6 +7,9 @@ namespace PlayFab
     {
         UnityWww, // High compatability Unity api calls
         HttpWebRequest, // High performance multi-threaded api calls
+#if UNITY_2017_1_OR_NEWER
+        UnityHttp, // Modern unity HTTP component
```
Comment (pgilmorepf) [inline targets/unity-v2/templates/PlayFabSettings.cs.ejs]: Lets rename this to "UnityWebRequest"

File: targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs [hunk 3]
```diff
@@ -0,0 +1,205 @@
+using System;
+using System.Collections;
+using System.IO;
+using PlayFab.ClientModels;
```
Comment (pgilmorepf) [inline targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs]: Delete ClientModels using.  (about a dozen reasons)

File: targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+#if !UNITY_WSA && !UNITY_WP8
+using Ionic.Zlib;
+#endif
+
+namespace PlayFab.Internal
+{
+#if UNITY_2017_1_OR_NEWER
+
+    using UnityEngine.Networking;
```
Comment (pgilmorepf) [inline targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs]: I'm not a fan of separating the usings into two places. Move this up.

File: targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+#endif
+
+            // Start the www corouting to Post, and get a response or error which is then passed to the callbacks.
+            PlayFabHttp.instance.StartCoroutine(Post(reqContainer));
+        }
+
+        private IEnumerator Post(CallRequestContainer reqContainer)
+        {
+#if PLAYFAB_REQUEST_TIMING
+                var stopwatch = System.Diagnostics.Stopwatch.StartNew();
```
Comment (pgilmorepf) [inline targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs]: Incorrect tabbing on stopwatch

File: targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+                StartTimeUtc = startTime,
+                ApiEndpoint = reqContainer.ApiEndpoint,
+                WorkerRequestMs = (int)stopwatch.ElapsedMilliseconds,
+                MainThreadRequestMs = (int)stopwatch.ElapsedMilliseconds
+            };
+            PlayFabHttp.SendRequestTiming(timing);
+#endif
+
+            if (!string.IsNullOrEmpty(www.error))
+                OnError(www.error, reqContainer);
```
Comment (pgilmorepf) [inline targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs]: Bracket inconsistency. If there are brackets on the if/else, then the whole if-switch should have brackets.  Don't mix.

File: targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+            return _pendingWwwMessages;
+        }
+
+
+        public void OnResponse(string response, CallRequestContainer reqContainer)
+        {
+            try
+            {
+#if PLAYFAB_REQUEST_TIMING
+                    var startTime = DateTime.UtcNow;
```
Comment (pgilmorepf) [inline targets/unity-v2/source/Shared/Internal/PlayFabHttp/PlayFabUnityHttp.cs]: Incorrect tabbing on startTime

**Comment by nitreo** [comment]
Introduced the changes

**Pull request merged**